src/js/portConnector.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2022, Tony Atkins
+ * Copyright 2011-2020, Colin Clark
+ *
+ *  Licensed under the MIT license, see LICENSE for details.
+ */
+
+(function (fluid) {
+    "use strict";
+    var youme = fluid.registerNamespace("youme");
+
+    fluid.defaults("youme.portConnector", {
+        gradeNames: ["fluid.modelComponent"],
+
+        events: {
+            onAccessError: null,
+            onAccessGranted: null,
+            onPortOpen: null
+        },
+
+        resources: {
+            system: {
+                promiseFunc: "{youme.system}.onCreate"
+            }
+        },
+
+        model: {
+            open: true,
+            portSpec: {},
+            connectionPort: null,
+
+            // Ensure that we're not created before our youme.system is ready.
+            system: "{that}.resources.system.parsed"
+        },
+
+        components: {
+            midiSystem: {
+                type: "youme.system",
+                options: {
+                    listeners: {
+                        "onAccessError.relay": "{youme.portConnector}.events.onAccessError.fire",
+                        "onAccessGranted.relay": "{youme.portConnector}.events.onAccessGranted.fire"
+                    }
+                }
+            }
+        },
+        dynamicComponents: {
+            connection: {
+                source: "{that}.model.connectionPort",
+                type: "youme.connection",
+                options: {
+                    model: {
+                        port: "{source}",
+                        open: "{youme.portConnector}.model.open"
+                    },
+                    listeners: {
+                        "onPortOpen.notifyParent": "{youme.portConnector}.events.onPortOpen.fire"
+                    }
+                }
+            }
+        },
+        modelListeners: {
+            "ports": {
+                // excludeSource: "init",
```

Comment:
One of these is commented out and the other not, presumably both can go. I think this could be replaced by a relay just as we can in multiPortConnector

Reply:
I knew both of them would be set on startup and wanted to avoid a double bump, but either a multi-path listener or relay sound like better options.

Reply:
I realised before posting the question about multi-path model relays that model relays run whenever any of their args are updated, so I was able to make this work.